core/src/main/java/org/apache/calcite/rel/metadata/RelMdPredicates.java

```diff
@@ -774,8 +774,6 @@ public RelOptPredicateList inferPredicates(
       case INNER:
       case LEFT:
       case ANTI:
-      case ASOF:
-      case LEFT_ASOF:
```

Comment:
So in these two situations, should an exception be thrown, or do we need to take other measures?

Reply:
It will return EMPTY in the end, looks good.

Reply:
> It will return EMPTY in the end, looks good.

thanks